scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoFinalize.scala

```diff
@@ -0,0 +1,33 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scala.meta.contrib._
+
+import metaconfig.{Conf, Configured}
+import scalafix.rule.Rule
+import scalafix.rule.RuleCtx
+import scalafix.lint.LintMessage
+import scalafix.lint.LintCategory
+
+case object NoFinalize extends Rule("NoFinalize") {
+  override def check(ctx: RuleCtx): Seq[LintMessage] = {
+    ctx.tree.collect {
+      case defn: Defn.Def if isFinalized(defn) => error(defn.name.pos)
+    }
+  }
+
+  private val errorCategory: LintCategory =
+    LintCategory.error(
+      "finalizer may be never called and have a severe performance penalty")
```

Comment:
> finalize may never get called and it incurs a performance penalty

Go any advice on what should the programmer should do instead? Maybe link to blog under `explain = "..."`